camelot/view/controls/delegates/customdelegate.py

```diff
@@ -218,13 +220,24 @@ def setEditorData(self, editor, index):
         editor.set_value(value)
 
         # update actions
-        action_states = index.model().data(index, ActionStatesRole)
-        if action_states is None:
-            return
-        for action_widget in editor.findChildren(ActionToolbutton):
-            if action_widget.action_route in action_states:
-                state = action_states[action_widget.action_route]
-                action_widget.set_state_v2(state)
+        if isinstance(editor, (Many2OneEditor, ChoicesEditor, DbImageEditor, FloatEditor, FileEditor)):
+            action_states = index.model().data(index, ActionStatesRole)
+            if action_states is None:
+                return
+            for action_widget in editor.findChildren(ActionToolbutton):
+                if action_widget.action_route in action_states:
+                    state = action_states[action_widget.action_route]
+                    action_widget.set_state_v2(state)
+        if isinstance(editor, One2ManyEditor):
```

Comment:
ik weet ook niet of dit stuk vd code geaccess w, gezien de One2ManyDelegate zijn eigen 'setEditorData' heeft ?

Reply:
Klopt, deze code had ik al verplaatst.